### PR TITLE
Adding ARHO manual

### PR DESCRIPTION
**Pull request by dhoyle-rh**
Adding Azure Red Hat OpenShift manual

### CODE REVIEW

**Comment by vikram-redhat** [comment]
Thanks @dhoyle-rh . As discussed, changes to _topic_map.yml should be in a separate PR in the 3.11 branch.

File: getting_started/access_your_services.adoc [hunk 1]
```diff
@@ -59,7 +59,46 @@ https://logs.<cluster-id>.openshift.com
...
+[[access-your-cluster-overview]]
+== Access your cluster
+
+// TODO Jim Minter is to provide updated URL info for ARHO
+
+Once your {product-title} cluster is configured and ready to use, you can access
+it through the following paths:
+
+* *Cluster ID*: The unique cluster name provided by the customer during provisioning. It is lowercase, and only contains letters, numbers, and hyphens.
+
```
Comment (y-cote) [inline getting_started/access_your_services.adoc]: * *Cluster Region*: The region in which the resources have been create in. Current list includes :  australiaeast, canadacentral, canadaeast, eastus, westus, westeurope, northeurope.

File: getting_started/access_your_services.adoc [hunk 2]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+----
+$ oc describe svc lb-service
+Name:                     lb-service
+Namespace:                default
+Labels:                   app=httpd-example
+Annotations:              <none>
+Selector:                 name=httpd-example
+Type:                     LoadBalancer
+IP:                       10.120.182.252
+LoadBalancer Ingress:     a5387ba36201e11e9ba901267fd7abb0-1406434805.us-east-1.elb.amazonaws.com
```
Comment (asalkeld) [inline getting_started/access_your_services.adoc]: azure named loadbalancer
you should probably go through that example and make new output for things like this

File: getting_started/access_your_services.adoc [hunk 3]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+$ oc expose dc httpd-example --type=LoadBalancer --name=lb-service
+service/lb-service exposed
+----
+
+=== Create an Internal-Only TCP Service
+
+You can alternatively expose your applications internally only, enabling access only through AWS VPC Peering or a VPN connection.
+
+----
+$ oc expose dc httpd-example --type=LoadBalancer --name=internal-lb --dry-run -o yaml | awk '1;/metadata:/{ print "  annotations:\n    service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0" }' | oc create -f -
```
Comment (asalkeld) [inline getting_started/access_your_services.adoc]: change
service.beta.kubernetes.io/aws-load-balancer-internal
to
service.beta.kubernetes.io/azure-load-balancer-internal

i checked that
git grep load-balancer-in
test/e2e/standard/cluster.go:           "service.beta.kubernetes.io/azure-load-balancer-internal": "true",

File: getting_started/access_your_services.adoc [hunk 4]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+
+You can expose your applications over an external LoadBalancer service, enabling access over the public Internet.
+
+----
+$ oc expose dc httpd-example --type=LoadBalancer --name=lb-service
+service/lb-service exposed
+----
+
+=== Create an Internal-Only TCP Service
+
```
Comment (asalkeld) [inline getting_started/access_your_services.adoc]: AWS -> Azure

File: getting_started/access_your_services.adoc [hunk 5]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+ifdef::openshift-arho[]
+[[expose-tcp-services]]
+== Expose TCP Services
+
+// TODO Need to update this with the Azure-specific load balancer info
+
+{product-title} xref:../architecture/networking/routes.adoc#architecture-core-concepts-routes[routes] expose applications by proxying traffic through HTTP/HTTPS(SNI)/TLS(SNI)
+to xref:../architecture/core_concepts/pods_and_services.adoc#architecture-core-concepts-pods-and-services[pods and services]. A link:https://kubernetes.io/docs/concepts/services-networking/#loadbalancer[LoadBalancer]
+service creates an AWS Elastic Load Balancer (ELB) for your {product-title} cluster, enabling direct TCP access to applications exposed by your LoadBalancer service.
```
Comment (asalkeld) [inline getting_started/access_your_services.adoc]: AWS -> Azure

Comment (y-cote) [inline getting_started/access_your_services.adoc]: AWS Elastic Load Balancer (ELB) -> Azure Load Balancer

File: getting_started/access_your_services.adoc [hunk 6]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+
+----
+$ oc expose dc httpd-example --type=LoadBalancer --name=lb-service
+service/lb-service exposed
+----
+
+=== Create an Internal-Only TCP Service
+
+You can alternatively expose your applications internally only, enabling access only through AWS VPC Peering or a VPN connection.
+
```
Comment (y-cote) [inline getting_started/access_your_services.adoc]: Not sure how this would translate

File: getting_started/access_your_services.adoc (hunk 3 above)
Comment (y-cote) [inline getting_started/access_your_services.adoc]: service.beta.kubernetes.io/aws-load-balancer-internal -> service.beta.kubernetes.io/azure-load-balancer-internal

File: getting_started/access_your_services.adoc [hunk 7]
```diff
@@ -59,7 +59,46 @@ https://logs.<cluster-id>.openshift.com
...
 ----
+endif::[]
+
+ifdef::openshift-arho[]
+[[access-your-cluster-overview]]
+== Access your cluster
+
+// TODO Jim Minter is to provide updated URL info for ARHO
+
+Once your {product-title} cluster is configured and ready to use, you can access
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: Need to say that all of the URLs include a cluster ID component (chosen by the service) and a region component (where the user provisioned the cluster)

File: getting_started/access_your_services.adoc (hunk 1 above)
Comment (jim-minter) [reply getting_started/access_your_services.adoc]: Don't list the regions in the documentation as they will change all the time

File: getting_started/access_your_services.adoc [hunk 8]
```diff
@@ -59,7 +59,46 @@ https://logs.<cluster-id>.openshift.com
 ----
 https://api.<cluster-id>.openshift.com/oauth2callback/<IdP name>
 ----
+endif::[]
+
+ifdef::openshift-arho[]
+[[access-your-cluster-overview]]
+== Access your cluster
+
+// TODO Jim Minter is to provide updated URL info for ARHO
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: provided below

File: getting_started/access_your_services.adoc [hunk 9]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+
+// TODO Need to update this with the Azure-specific load balancer info
+
+{product-title} xref:../architecture/networking/routes.adoc#architecture-core-concepts-routes[routes] expose applications by proxying traffic through HTTP/HTTPS(SNI)/TLS(SNI)
+to xref:../architecture/core_concepts/pods_and_services.adoc#architecture-core-concepts-pods-and-services[pods and services]. A link:https://kubernetes.io/docs/concepts/services-networking/#loadbalancer[LoadBalancer]
+service creates an AWS Elastic Load Balancer (ELB) for your {product-title} cluster, enabling direct TCP access to applications exposed by your LoadBalancer service.
+
+[NOTE]
+====
+LoadBalancer services require an additional purchase. Contact your sales team if you are interested in using LoadBalancer services for your {product-title} cluster.
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: "additional purchase" is not applicable to Azure

File: getting_started/access_your_services.adoc [hunk 10]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+====
+
+=== Check your LoadBalancer Quota
+
+By purchasing LoadBalancer services, you are provided with a quota of LoadBalancers available for your {product-title} cluster.
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: not applicable

File: getting_started/access_your_services.adoc (hunk 4 above)
Comment (jim-minter) [reply getting_started/access_your_services.adoc]: "AWS VPC Peering or a VPN connection" -> Azure VNet peering

File: getting_started/access_your_services.adoc [hunk 11]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+You can alternatively expose your applications internally only, enabling access only through AWS VPC Peering or a VPN connection.
+
+----
+$ oc expose dc httpd-example --type=LoadBalancer --name=internal-lb --dry-run -o yaml | awk '1;/metadata:/{ print "  annotations:\n    service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0" }' | oc create -f -
+service/internal-lb exposed
+----
+
+=== Use your TCP Service
+
+Once your LoadBalancer service is created, you can access your service by using the URL provided to you by {product-title}.
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: URL -> IP

File: getting_started/access_your_services.adoc [hunk 12]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+The `LoadBalancer Ingress` value is a URL unique to your service that remains static as long as the service is not deleted.
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: You can specify an Azure DNS label for the service using the `service.beta.kubernetes.io/azure-dns-label-name` annotation

File: getting_started/access_your_services.adoc [hunk 13]
```diff
@@ -180,17 +223,94 @@ Endpoints:                <none>
...
+Endpoints:                <none>
+Session Affinity:         None
+External Traffic Policy:  Cluster
+----
+endif::[]
 
 [[access-your-cluster-status-updates]]
 == Receive status updates
 
+// TODO Need to check to see if Microsoft is providing this
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: Overall service status/health information is available at https://azure.microsoft.com/en-us/status/

File: getting_started/access_your_services.adoc [hunk 14]
```diff
@@ -215,6 +335,8 @@ xref:../admin_guide/quota.adoc#admin-guide-quota[managing
 quotas] and xref:../admin_guide/service_accounts.adoc#admin-guide-service-accounts[configuring service
 accounts].
 
+// TODO Need to confirm if the following paragraph is true for ARHO
```
Comment (jim-minter) [inline getting_started/access_your_services.adoc]: On OSA all clusters are configured using the NetworkPolicy SDN.
Project admins can set NetworkPolicy objects on their projects.
Cluster admins can also set EgressNetworkPolicies

File: getting_started/dedicated_administrators.adoc [hunk 15]
```diff
@@ -235,6 +248,9 @@ resources and objects are consumed by users across the cluster.
 Defaults can be set for quotas and limit ranges for new projects at creation. To
 request such a change, open a support case on the
 https://access.redhat.com/support/[Red Hat Customer Portal].
+endif::[]
+
+// TODO Need to add a section describing how Dedicated Admins can manage egress network policies.
```
Comment (jim-minter) [inline getting_started/dedicated_administrators.adoc]: need to make a decision on whether you're calling it dedicated admin or customer admin

Comment (jim-minter) [reply getting_started/dedicated_administrators.adoc]: link to https://docs.openshift.com/container-platform/3.11/admin_guide/managing_networking.html#admin-guide-controlling-egress-traffic

**Comment by dhoyle-rh** [comment]
Accidentally closed this PR -- reopening it.

File: getting_started/access_your_services.adoc (hunk 2 above)
Comment (dhoyle-rh) [reply getting_started/access_your_services.adoc]: @grantomation, can you provide the new output for the Azure Load Balancer? Or maybe we can just link to Azure docs?

Comment (grantomation) [reply getting_started/access_your_services.adoc]: It is difficult to find context for this as the commits are missing. If I go to the renedered version of @dhoyle-rh's branch I can see a few things on this page that require attention. I have put my comments in the google doc that I emailed around yesterday. 

If the loadbalancer service is meant to be created from the command `$ oc expose dc httpd-example --type=LoadBalancer --name=internal-lb --dry-run -o yaml | awk '1;/metadata:/{ print "  annotations:\n    service.beta.kubernetes.io/azure-load-balancer-internal: 0.0.0.0/0" }' | oc create -f -` in the previous section "Create an Internal-Only TCP Service", then the name is wrong and this command should be `oc describe svc internal-lb` instead. If that is the case, then the output will look something like follows;

```
$ oc describe svc internal-lb
Name:                     internal-lb
Namespace:                cake
Labels:                   app=cakephp-mysql-persistent
                          template=cakephp-mysql-persistent
                          template.openshift.io/template-instance-owner=3a53eabd-6b11-11e9-948c-000d3ab9a700
Annotations:              service.beta.kubernetes.io/azure-load-balancer-internal=0.0.0.0/0
Selector:                 name=cakephp-mysql-persistent
Type:                     LoadBalancer
IP:                       172.30.166.170
LoadBalancer Ingress:     52.155.223.79
Port:                     <unset>  8080/TCP
TargetPort:               8080/TCP
NodePort:                 <unset>  30823/TCP
Endpoints:                10.128.2.9:8080
Session Affinity:         None
External Traffic Policy:  Cluster
Events:
  Type    Reason                Age   From                Message
  ----    ------                ----  ----                -------
  Normal  EnsuringLoadBalancer  24m   service-controller  Ensuring load balancer
  Normal  EnsuredLoadBalancer   24m   service-controller  Ensured load balancer
```

However there are a couple of differences. I have used the name "internal-lb" as was created by the previous command and I am using the cake-php example in a project/namespaces that is not "default". We probably don't need to include the events. I hope this helps.

**Comment by adellape** [comment]
@mgahagan73 @e-tienne Looking for QE review on this new doc set for ARO; could you take a look? In particular, the files changed in this PR, but anything else ARO-specific you might see elsewhere that you think still needs changing. These docs are based off of the latest OSD docs.

Here's an internal preview:

http://file.rdu.redhat.com/~adellape/050119/arho/welcome/index.html

**Comment by adellape** [comment]
@jim-minter FYI: see above for preview build of latest changes.

**Comment by adellape** [comment]
Close this PR in favor of https://github.com/openshift/openshift-docs/pull/14652, which takes this commit and continues with further edits:

https://github.com/openshift/openshift-docs/pull/14652

Please provide feedback there.

**Pull request closed without merging**